www/js/common/services.js

```diff
@@ -208,7 +209,66 @@ angular.module('emission.main.common.services', [])
           "features": places.concat(trips)
         };
     };
+    var getDisplayNameForTrip = function(common_trip) {
```

Comment:
Should just change `getDisplayName` to take a `cobject`?
Or even to return a promise that you use to set the value for whatever object.
But you can fix this later.